core/include/kernel/dt.h

```diff
@@ -203,6 +203,7 @@ static inline int _fdt_get_status(const void *fdt __unused, int offs __unused)
 	return -1;
 }
 
+__noreturn
```

Comment:
A minor note (I'm OK with this too):
I find it a bit odd to have `__noreturn` on one line and then `static inline void` on another. I would prefer something like:
``` C
__noreturn static inline void
_fdt_fill_device_info(void *fdt __unused, struct dt_node_info *info __unused,
                      int node __unused)
```

Reply:
Agreed, but I chose the lone `__noreturn` for better grep-ability.
If you grep for `_fdt_fill_device_info` it's still obvious this match is the stub, because you get `static inline void _fdt_fill_device_info`. With your proposal it stands out a bit less.